### ISSUE TITLE
Switch to Next.js

### ISSUE THREAD

**Issue by siddiquiaffan** [moogle-cc/moogle-blog#18]
Next.js is a react framework for production.

It has some intresting features listed down below:
- SEO Friendly
- Image Optimization
- Server side rendering
- File system routing
And many more.

Head over to https://nextjs.org/ for more.

**Comment by saiorama** [comment]
I looked at next.js and decided not to use it because it needs pre-processing to generate static files. It's not well suited for fully serverless blogs like Moogle. Happy to be corrected if you know something about next.js that I don't.

**Comment by siddiquiaffan** [comment]
Not exactly. You don't need to generate static files. NextJs supports dynamic rendering. Have a look at this https://nextjs.org/docs/basic-features/data-fetching.

For example if you want that, data should be fetched from server on every request, next.js has `getServerSideProps`.
If you want to revalidate after a certain time next.js has `getStaticProps`.

Next.js only generates static files for those pages in which no dynamic data is fetched from server or database. For example Home page, About page etc. Even you can use `getStaticProps` on those pages so that it'll revalidate after certain time as data might not be going to change/update very frequently.

BTW it's your choice 🙂

**Comment by siddiquiaffan** [comment]
In case you have any further doubt you can connect me on telegram or via email.
